### PR TITLE
Remove unnecessary flake8 exceptions

### PR DESCRIPTION
**Pull request by bryant1410**
Also tackles https://github.com/allenai/allennlp/pull/3760#issuecomment-584709526 and https://github.com/allenai/allennlp/pull/3759#discussion_r377721012

### CODE REVIEW

**Comment by matt-gardner** [comment]
@DeNeutoy, just using `AllenNlpTestCase` should do the necessary imports, right?  I'm not sure why we need the flake exception.  There are only 6 failing tests here, and I think they are failing because of missing imports in the corresponding `__init__.py` files, so this is actually catching bugs.  Also, I think we should avoid using `from_params` to construct things in tests going forward, except for unusual cases, or integration-style tests like our model tests.

On reshuffling of imports: when I touch a file, I will often sort the imports in that file - makes it easier to find things later.  I didn't look closely at all of the shuffling in this PR, but it looks like that's all that's going on here.

**Comment by DeNeutoy** [comment]
Oh ok, happy to take your lead on both issues

**Comment by bryant1410** [comment]
> @DeNeutoy, just using `AllenNlpTestCase` should do the necessary imports, right? I'm not sure why we need the flake exception. There are only 6 failing tests here, and I think they are failing because of missing imports in the corresponding `__init__.py` files, so this is actually catching bugs. Also, I think we should avoid using `from_params` to construct things in tests going forward, except for unusual cases, or integration-style tests like our model tests.

I'm trying to add the missing imports.

The `from_params` is used in a script. In any case, IIUC, @DeNeutoy you say you're gonna fix this?

> On reshuffling of imports: when I touch a file, I will often sort the imports in that file - makes it easier to find things later. I didn't look closely at all of the shuffling in this PR, but it looks like that's all that's going on here.

I think we can add a linter to optimize the imports, like in the transformers library: https://github.com/huggingface/transformers/blob/ee5de0ba449d638da704e1c03ffcc20a930f5589/.circleci/config.yml#L105

File: .flake8 [hunk 1]
```diff
@@ -12,41 +12,14 @@ exclude =
     tutorials/tagger/**
 
 per-file-ignores =
-    # __init__.py files are allowed to have unused imports and lines-too-long
-    allennlp/__init__.py:F401
-    allennlp/**/__init__.py:F401,E501
+    # __init__.py files are allowed to have unused imports
+    */__init__.py:F401
+    */**/**/__init__.py:F401
```
Comment (bryant1410) [inline .flake8]: Apparently this works. If you can come up with a way to grab all with one line, that'd be cool.

File: allennlp/commands/elmo.py [hunk 2]
```diff
@@ -68,25 +68,24 @@
...
-from allennlp.common.checks import ConfigurationError
 from allennlp.data.token_indexers.elmo_indexer import ELMoTokenCharactersIndexer
-from allennlp.nn.util import remove_sentence_boundaries
 from allennlp.modules.elmo import _ElmoBiLm, batch_to_ids
-from allennlp.commands.subcommand import Subcommand
+from allennlp.nn.util import remove_sentence_boundaries
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=FutureWarning)
+    import h5py
```
Comment (bryant1410) [inline allennlp/commands/elmo.py]: I put it at the end so the regular imports are sorted.

File: allennlp/common/util.py [hunk 3]
```diff
@@ -434,7 +434,7 @@ def push_python_path(path: PathType) -> ContextManagerFunctionReturnType[None]:
         sys.path.remove(path)
 
 
-def import_submodules(package_name: str) -> None:
+def import_module_and_submodules(package_name: str) -> None:
```
Comment (bryant1410) [inline allennlp/common/util.py]: Note I make this change mentioned in another PR here.

File: scripts/convert_openie_to_conll.py [hunk 4]
```diff
@@ -217,7 +217,7 @@ def read(fn: str) -> Iterable[List[Extraction]]:
         yield prev_sent
 
 
-def convert_sent_to_conll(sent_ls: List[Extraction]) -> Iterable[Tuple[str, str, ...]]:
+def convert_sent_to_conll(sent_ls: List[Extraction]) -> Iterable[Tuple[str, ...]]:
```
Comment (bryant1410) [inline scripts/convert_openie_to_conll.py]: I changed this because in the end you can only put ellipsis in the second arg.

**Comment by bryant1410** [comment]
Tests pass now, but I'm not quite sure where what was missing gets imported exactly.

Also, I believe that tests that load archives with config files can locally import what's necessary inside the test function (or `setUp`).

File: .flake8 (hunk 1 above)
Comment (matt-gardner) [reply .flake8]: Please leave the `lines-too-long`; black will catch anything that's fixable, so we don't need flake to complain at us for these files.  Even if there aren't any files where it matters now, it'd be annoying to have this come up later.

**Comment by bryant1410** [comment]
About messing with `sys.path` within the scripts, is it really necessary? If they're called from the root dir they should work well.

**Comment by matt-gardner** [comment]
It may not be necessary, but I and others have run into it enough times that it's convenient to have, and it doesn't cause problems.
